src/stages/models/manager.ts

```diff
@@ -219,4 +226,24 @@ export class StageManager {
 
     return stage;
   }
+
+  /**
+   * This method is used to update the progress of a stage according tasks metrics.
+   * @param stageId unique identifier of the stage.
+   */
+  public async updateStageProgress(stageId: string, jobId: string): Promise<void> {
+    // calculate stage progress data
+    const currentSummary = await aggregateTaskStatusesForStage(stageId, this.prisma);
+    const completionPercentage = getCurrentPercentage(currentSummary);
+    let stageUpdatedData: Prisma.StageUpdateInput = { summary: currentSummary, percentage: completionPercentage };
+
+    const job = await this.jobManager.getJobById(jobId);
+
+    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
+    if (completionPercentage === 100 && job.jobMode === JobMode.PRE_DEFINED) {
```

Comment:
change it to total === completed when possible. (maybe add comment about it)

Reply:
done